### ISSUE TITLE
Timestamp format customization through config?

### ISSUE THREAD

**Issue by MicMicMon** [dbfannin/ngx-logger#96]
Feature request: being able to give a format for the timestamp in the config object.

When in development environment, you don't need to output the date, and it is clearer if timezone can be set to local.

Do you also think it would be a useful development, instead of having only ISO Date format?

Thanks for the great library!

**Comment by dbfannin** [comment]
@MicMicMon, I'm not sure I'll be able to get around to this any time soon. This has been requested before, so I think it would be really valuable to have. If you wanted to take a stab at it, I'd be more than happy to take a look at the PR!

**Comment by noririco** [comment]
var tzoffset = (new Date()).getTimezoneOffset() * 60000; //offset in milliseconds
var localISOTime = (new Date(Date.now() - tzoffset)).toISOString().slice(0, -1);

what about this  in the logger service : var timestamp = ......

**Comment by dbfannin** [comment]
agreed with @MicMicMon. And I wouldn't want to include a date library for this. It would make the package much larger than it needs to be for very little benefit. 

There are 2 possible solutions that I think would work. both with their Pros and Cons.

- Allow for a finite set of formats (such as "ShortDateTime" "LongDateTime") and allow the user to also set the timezone offset in the config. 
   - Con: The user will have to adjust the timezone offset for DST since -6 is -5 during DST
   - Pro: We handle the formatting for the user, one less thing they have to worry about

- Create a hook to allow custom hook to allow users to create and register their own date/time formatter.
  - Something like this: `registerDateTimeFormatter(formatter: (date: Date) => string) {}`  (see how we register the custom log monitor) 
  - Pro: We don't have to maintain any date formatting code.Uusers can format their date strings however they would like
  - Con: The user has to learn how to register their formatter, write their own custom code for formatting their date, and could still get caught in the DST bug

**Comment by MicMicMon** [comment]
Implemented the second solution in the config object in https://github.com/dbfannin/ngx-logger/pull/159
